deploy.gradle

```diff
@@ -113,8 +113,8 @@ subprojects {
         // upload to the releases repository using ./gradlew -Prelease publish
         def apacheSnapshotsRepoUrl = 'https://repository.apache.org/content/repositories/snapshots'
         def apacheReleasesRepoUrl = 'https://repository.apache.org/service/local/staging/deploy/maven2'
-        def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? project.property('mavenSnapshotsRepo') : "$apacheSnapshotsRepoUrl"
-        def releasesRepoUrl = project.hasProperty('mavenReleasesRepo') ? project.property('mavenReleasesRepo') : "$apacheReleasesRepoUrl"
+        def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? "$mavenSnapshotsRepo" : "$apacheSnapshotsRepoUrl"
```

Comment:
I'm a little lost here, it looks like previously we could set this based on the -PmavenSnapshotsRepo and now we are switching it to a local property? Or does -P handle set the local property as well?

Mostly i'm confused why project.property was not sufficient before

Reply:
Groovy and Gradle still hold a lot of mysteries for me

Reply:
From my understanding project.property('mavenSnapshotsRepo') picks up properties from either a gradles.properties file or from -P args, I"ll run some tests to see

Reply:
yeap, and -P args should have the high priority.

Reply:
emm... github webpage looks good to me.

Reply:
oh, i see. the inlined conversation diff looks weird, but `File Changed` page looks good.

Reply:
I'm still lost a bit I think
```groovy
  def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? "$mavenSnapshotsRepo" : "$apacheSnapshotsRepoUrl"
```

should be 
```groovy
  def snapshotsRepoUrl = project.hasProperty('mavenSnapshotsRepo') ? project.property("mavenSnapshotsRepo") : "$apacheSnapshotsRepoUrl"
  ```
  
  When I do a quick test
  
```groovy
print(project.hasProperty("mavenSnapshotsRepo"))
print(project.property("mavenSnapshotsRepo"))
print($mavenSnapshotsRepo)
```

and
```bash
 g -PmavenSnapshotsRepo=one
```

Then I get mavenSnapshotsRepo undefined but the first two work fine.

Reply:
`print("$mavenSnapshotsRepo")` works but `print($mavenSnapshotsRepo)` not.

"$mavenSnapshotsRepo" will fallback to `project.property("mavenSnapshotsRepo")`

Reply:
So this is some magic in the string interpolation?

Reply:
yes